config/groovy/common.groovy

```diff
@@ -408,24 +476,116 @@ class common {
+            if (!(excludedItems.contains(itemName))) {
                 localItems << itemName
             }
         }
         return localItems
     }
 
     void cacheItemList() {
-        cachedItemList = retrieveAvailableItems()
+        if (!itemListCached) {
+            cachedItemList = retrieveAvailableItems()
+        }
         itemListCached = true
     }
 
     void unCacheItemList() {
         itemListCached = false
     }
+
+    void writeDependencyDotFileForModule(File dependencyFile, File module) {
+        if (module.name.contains(".")) {
+            println "\"" + module.name + "\" is not a valid source (non-jar) module - skipping"
+        } else if (itemsRetrieved.contains(module.name)) {
+            println "Module \"" + module.name + "\" was already handled - skipping"
+        } else if (!module.exists()) {
+            println "Module \"" + module.name + "\" is not locally available - skipping"
+            itemsRetrieved << module.name
+        } else {
+            def foundDependencies = itemTypeScript.findDependencies(module, false)
+            if (foundDependencies.length == 0) {
+                // if no other dependencies exist, depend on engine
+                dependencyFile.append("  \"" + module.name + "\" -> \"engine\"\n")
+                itemsRetrieved << module.name
+            } else {
+                // add each of $foundDependencies as item -> foundDependency lines
+                for (dependency in foundDependencies) {
+                    dependencyFile.append("  \"" + module.name + "\" -> \"$dependency\"\n")
+                }
+                itemsRetrieved << module.name
+
+                // find dependencies to progress with
+                String[] uniqueDependencies = foundDependencies - itemsRetrieved
+                if (uniqueDependencies.length > 0) {
+                    for (dependency in uniqueDependencies) {
+                        writeDependencyDotFileForModule(dependencyFile, new File("modules/$dependency"))
+                    }
+                }
+            }
+        }
+    }
+
+    def refreshGradle() {
+        def localItems = []
+        targetDirectory.eachDir() { dir ->
+            String itemName = dir.getName()
+            // Don't consider excluded items
+            if (!(excludedItems.contains(itemName))) {
+                localItems << itemName
+            } else {
+                println "Skipping $dir as it is in the exclude list"
+            }
+        }
+
+        for (String item : localItems) {
+            itemTypeScript.refreshGradle(new File(targetDirectory, item))
+        }
+    }
+}
+
+/**
+ * Small ANSI coloring utility.
+ * @see https://gist.github.com/tvinke/db4d21dfdbdae49e6f92dcf1ca6120de
+ * @see http://www.bluesock.org/~willg/dev/ansi.html
+ * @see https://gist.github.com/dainkaplan/4651352
+ */
+class Ansi {
+
+    static final String NORMAL         = "\u001B[0m"
+
+    static final String BOLD           = "\u001B[1m"
```

Comment:
I've had issues with rendering ANSI escape codes on Windows before but I can't find any issues while testing right now, so this should be alright, hopefully.

Reply:
This works under the Windows Terminal app but doesn't work under a plain Command Prompt. It just causes cosmetic issues though.

Reply:
There's a whole new system for colors with the PicoCLI approach in the new GooeyCLI system, which works nicely on Windows, so no worries there. Probably lovely _new_ worries in the new system though, when we get there! :-)